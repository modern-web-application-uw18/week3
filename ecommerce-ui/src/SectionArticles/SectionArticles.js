import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Card from '../Card/Card.js';

export default class SectionArticles extends Component {

    render() {
      const articles = this.props.articles;
      const newArticles = articles.map((article, index) => {
        return <Card article={article} title={article.title} image={article.image} key={index} index={index}/>
      });

        return (
          <section className="articles">
            <h2>Plan An Event</h2>
            {newArticles}
          </section>
        );
    }
}

SectionArticles.propTypes = {
  articles: PropTypes.array.isRequired
}

SectionArticles.defaultProps = {
  articles: [{
    title: 'test_title',
    image: 'test_url',
    description: 'this is a test description',
    experiences: {
      type: 'Mind Exploding',
      amount: 1
    }
  }]
}
