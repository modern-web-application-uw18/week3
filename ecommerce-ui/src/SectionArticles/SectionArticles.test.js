import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';

import SectionArticles from './SectionArticles.js';

describe('SectionArticles', () => {
  it('Smoke Test', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SectionArticles />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Match Snapshot', () => {
    const component = TestRenderer.create(<SectionArticles />);
    expect(component.toJSON()).toMatchSnapshot();
  });

})
