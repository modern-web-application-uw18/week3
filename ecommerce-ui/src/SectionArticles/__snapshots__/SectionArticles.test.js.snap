// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionArticles Match Snapshot 1`] = `
<section
  className="articles"
>
  <h2>
    Plan An Event
  </h2>
  <div
    className="card"
  >
    <div
      className="coverImage"
      style={
        Object {
          "background": "url('test_url') center / cover no-repeat",
        }
      }
    >
      <span
        className="textOverlay"
      >
        test_title
      </span>
    </div>
    <div
      className="cardInfo"
    >
      <div
        className="articleInfo"
      >
        <div
          className="title"
        >
          1
          + 
          Mind Exploding
        </div>
        <div
          className="text"
        >
          this is a test description
        </div>
      </div>
    </div>
  </div>
</section>
`;
